pgmpy/factors/CPD.py

```diff
@@ -116,15 +119,33 @@ def __init__(self, variable, variable_card, values,
 
         super(TabularCPD, self).__init__(variables, cardinality, values.flatten('C'))
 
-    def get_cpd(self):
+    def get_cpd(self, pretty=True):
         """
         Returns the cpd
         """
         if self.variable in self.variables:
+            if pretty:
+                table_list = []
+                for i in range(len(self.cardinality))[1:]:
+                    var_rep = np.prod(self.evidence_card[i:])
+                    row = [self.scope()[-i]]
+                    row.extend(self.variables[row[0]] * var_rep)
+                    if i > 1:
+                        row[1:] = [x for item in row[1:] for x in
+                                   repeat(item, int(np.prod(self.evidence_card[0:i - 1])))]
```

Comment:
You are adding values in `row` in Line 132 and then replacing them in line 134 except for one case when `i = 1`. There should be a generalized way to do this and if not you can atleast put an if-else.